ui/dashboards/src/components/PanelDrawer/PanelOptionsEditor.tsx

```diff
@@ -23,10 +23,14 @@ export interface PanelOptionsEditorProps {
 export function PanelOptionsEditor(props: PanelOptionsEditorProps) {
   const { kind, value, onChange } = props;
   const { OptionsEditorComponent, createInitialOptions } = usePanelPlugin(kind);
+  const [currentKind, setCurrentKind] = useState(kind);
 
   // When the kind changes, re-init options
   useEffect(() => {
-    onChange(createInitialOptions());
+    if (kind !== currentKind) {
+      onChange(createInitialOptions());
+      setCurrentKind(kind);
+    }
```

Comment:
Without this, this would overwrite the existing values (reset them to empty values), because the `useEffect` was called multiple times with "Markdown" as the same value for `kind`.